### PR TITLE
Fully handle "?" shortcut

### LINKED ISSUE #75386: Rustdoc: Help dialog doesn't open when the `?` key is not behind Shift
I'm using a [weird keyboard layout](https://neo-layout.org) which doesn't put the `?` key behind Shift, which stopped me from ever viewing the help popup in Rustdoc. If I had managed to open the help popup, I would have learned about the incredibly useful [type-based search feature](https://github.com/rust-lang/rust/issues/75197) much sooner. :+1: 

As far as I can tell, the code responsible is [here](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/main.js#L411-L413): It's checking for `ev.shiftKey` unconditionally, but I believe it's only supposed to check when [`ev.key` is undefined](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/main.js#L339).

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #75386.

cc @runiq

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -408,7 +408,7 @@ function defocusSearchBar() {
                 break;
 
             case "?":
-                if (ev.shiftKey) {
+                if (ev.key || ev.shiftKey) {
```
Comment (pickfire) [inline src/librustdoc/html/static/main.js:411]: Why not just remove this branch?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: I've been switching between this and removing the condition for a while. You made the balance change haha.

**Review by pickfire — APPROVED** [review]
Looks good to me but I didn't test it out.

**Comment by GuillaumeGomez** [comment]
I tested both and they worked on my computer. If an issue occurs, it's pretty simple to rollback.

@bors: r=pickfire rollup

**Comment by jyn514** [comment]
Looks like everyone took care of this without me 😆

**Pull request merged**